src/scene/light.js

```diff
@@ -452,7 +452,7 @@ class Light {
     }
 
     get shape() {
-        return this._shape;
+        return this._shape ? this._shape : LIGHTSHAPE_PUNCTUAL;
```

Comment:
Surely this is unnecessary if `_shape` is always guaranteed to be set to a valid value.

Reply:
I can test - but it doesn't seem that in the editor _shape will use class defaults if not present in the component - which is why editor migration code is needed

Reply:
tested and removed the redundant code in the get accessor